File: C++/find-the-winner-of-the-circular-game.cpp
class Solution {
public:
    int findTheWinner(int n, int k) {
        vector<int> p(n);
        for (int i = 0; i < n; i++) {
            p[i] = i + 1 ;
        }
        
        int x = 0;
        while (p.size() > 1) {
            x = (x + k - 1) % p.size();
            p.erase(p.begin() + x);
        }
        
        return p[0];
    }
};

File: Go/find-the-winner-of-the-circular-game.go
func findTheWinner(n int, k int) int {
	p := make([]int, n)
	for i := range p {
		p[i] = i + 1
	}

	x := 0
	for len(p) > 1 {
		x = (x + k - 1) % len(p)
		p = slices.Delete(p, x, x+1)
	}

	return p[0]
}